src/frontend/src/common/types/user.d.ts

```diff
@@ -1,17 +1,27 @@
 interface Group {
-  [index: string]: JSONPrimitive;
-  type: "V2Group";
   id: number;
   name: string;
 }
 
 interface User {
-  [index: string]: JSONPrimitive;
-  type: "User";
   id: number;
   name: string;
-  group: V2Group;
+  group: Group;
   agreedToTos: boolean;
   acknowledgedPolicyVersion: string | null; // Date or null in DB. ISO 8601: "YYYY-MM-DD"
   splitId: string;
 }
+
+interface RawGroupRequest {
```

Comment:
Nit: I would put the `RawBlah` interfaces in the same file as where the API calling functions are since I expect those are the only functions that need to know the interface. I think of the `*.d.ts` files as being more about defining common interfaces that are used throughout the codebase. If the interface/type is restricted to only being used in a couple files, it feels like we normally just define them in one and export/import. It's a bit fuzzy, and I'd defer to Maya's opinion, I've never really thought this much about it since I almost never add new types to the `*.d.ts` files

Reply:
I moved them for now!